src/builder.rs

```diff
@@ -51,11 +52,47 @@ use crate::values::{
 
 use crate::{AtomicOrdering, AtomicRMWBinOp, FloatPredicate, IntPredicate};
 
+use std::cell::Cell;
 use std::marker::PhantomData;
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum PositionState {
+    NotSet,
+    Set,
+}
+
+/// Errors that can be generated by the Builder. All `build_*` methods return a `Result<_, BuilderError>`, which must be handled.
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum BuilderError {
+    #[error("Builder position is not set")]
+    UnsetPosition,
+    #[error("GEP error: does not point to a struct or index is out of bounds")]
+    GEPError,
```

Comment:
This is called `GEPError`, in my head it goes like "any GEP error goes into this variant", but there are actually 2 other GEP error variants.

I'd say this one should have a different name or be broken down into more variants.

WDYT? Is this worth an issue?

Reply:
I removed the `GEPError` on my fork. Turns out, `GEPPointee` and `GEPIndex` are all that were needed.

Reply:
Should I reopen another PR?

Reply:
Oh nice, I'd say yes! But we can wait for TheDan to be 100% sure.